src/filebrowser_upload/main.py

```diff
@@ -21,63 +26,155 @@
 
 
 def get_args():
-    if '--version' in sys.argv:
+    """
+    Get arguments.
+
+    Returns:
+        args: Parsed arguments.
+    """
+    if "--version" in sys.argv:
         print(__version__)
         sys.exit(0)
-    parser = argparse.ArgumentParser(description='Filebrowser upload.')
-    parser.add_argument('filepath', help='local file path')
-    parser.add_argument('--version', dest='version', help='Show version')
-    parser.add_argument('--api', dest='api', required=True, help='Filebrowser upload API URL')
-    parser.add_argument('--username', dest='username', required=True, help='Username')
-    parser.add_argument('--password', dest='password', required=True, help='Password')
-    parser.add_argument('--dest', dest='dest', required=True, help='File destination')
-    parser.add_argument('--insecure', dest='insecure', action='store_true',
-                        default=False, help='Allow insecure server connections when using SSL')
-    parser.add_argument('--no-progress', dest='no_progress', action='store_true',
-                        default=False, help='Disable progress bar')
-    parser.add_argument('--override', dest='override', action='store_true',
-                        default=False, help='Override file or not')
+
+    parser = argparse.ArgumentParser(description="Filebrowser upload.")
+    parser.add_argument("--version", dest="version", help="Show version")
+    parser.add_argument(
+        "--api", dest="api", required=True, help="Filebrowser upload API URL"
+    )
+    parser.add_argument("--username", dest="username", required=True, help="Username")
+    parser.add_argument("--password", dest="password", help="Inline password")
+    parser.add_argument(
+        "--insecure",
+        dest="insecure",
+        action="store_true",
+        default=False,
+        help="Allow insecure server connections when using SSL",
+    )
+    parser.add_argument(
+        "--no-progress",
+        dest="no_progress",
+        action="store_true",
+        default=False,
+        help="Disable progress bar",
+    )
+    parser.add_argument(
+        "--override",
+        dest="override",
+        action="store_true",
+        default=False,
+        help="Override files or not",
+    )
+    parser.add_argument(
+        "--dry-run",
+        dest="dry_run",
+        action="store_true",
+        default=False,
+        help="Dry run mode (no upload)",
+    )
+
+    subparsers = parser.add_subparsers(
+        help="Commands for single file upload or entire folder", dest="subcommand"
+    )
+
+    # Parser for file command
```

Comment:
Seems not need subparser, can auto detect file or folder upload by check file type.

Reply:
Subparser removed